store-and-retrieve-data/use-storagehub-sdk/get-started.md

```diff
@@ -78,6 +78,139 @@ Add the core and MSP client packages to your project. These libraries provide th
+
+    ```ts title="// **PLACEHOLDER FOR STEP 2: CREATE POLKADOT API CLIENT**"
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/get-started/client-service.ts:polkadot-api-client'
+    ```
+
+--8<-- 'text/store-and-retrieve-data/use-storagehub-sdk/initialize-client-service-summary.md'
+
+??? code "View complete file"
+
+    ```ts title="clientService.ts"
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/get-started/client-service.ts'
+    ```
+
+### Set Up MSP Service
+
+To interact with DataHaven's Main Storage Provider (MSP) services, you need to establish a connection using the `MspClient` from the StorageHub SDK. This involves configuring the HTTP client, setting up session management for authenticated requests, and initializing the MSP client itself.
+
+
+1. Create a `mspService.ts` file within your `services` folder and add the following code:
+
+    !!! note
+        The code below uses **DataHaven Testnet** configuration values, which include the **Chain ID**, **RPC URL**, **WSS URL**, **MSP URL**, and **token metadata**. If you’re running a **local devnet**, make sure to replace these with your local configuration parameters. You can find all the relevant **local devnet values** in the [Quick Start Guide](/store-and-retrieve-data/quick-start).
+
+    ```ts title="mspService.ts"
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/get-started/msp-service.ts:imports'
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/get-started/msp-service.ts:initial-setup'
+  
+    // --- MSP Service logic ---
+    // **PLACEHOLDER FOR STEP 1: CONNECT MSP CLIENT **
+    // **PLACEHOLDER FOR STEP 2: CREATE MSP HELPER FUNCTIONS **
+
+    // Export initialized clients and accounts for reuse across the project.
+    // This allows other modules to easily import shared configuration and avoid duplicate setup.
+    export { mspClient, getMspInfo, getMspHealth, authenticateUser };
+    ```
+
+3. Connect to the MSP Client
+
+    Add the following code to connect to the MSP Client:
+
+    ```ts title="// **PLACEHOLDER FOR STEP 1: CONNECT MSP CLIENT**"
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/get-started/msp-service.ts:connect-msp-client'
+    ```
+
+4. Create MSP Helper Functions
+
+    Add the following code to create MSP helper functions regarding MSP info, MSP health, and user authentication via the MSP:
+
+    ```ts title="// **PLACEHOLDER FOR STEP 2: CREATE MSP HELPER FUNCTIONS**"
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/get-started/msp-service.ts:msp-helper-functions'
+    ```
+
+--8<-- 'text/store-and-retrieve-data/use-storagehub-sdk/initialize-msp-service-summary.md'
+
+??? code "View complete file"
+
+    ```ts title="mspService.ts"
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/get-started/msp-service.ts'
+    ```
+
 ## Next Steps
 
 Now that you have the StorageHub SDK packages installed, you are ready to start building with DataHaven. 
```

Comment:
maybe add something about clients set up?

Reply:
Not sure if I understand this one, since the Next Steps section is highlighted

Reply:
Okay, it clicked just now.